packages/compass-data-modeling/src/components/export-diagram-modal.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useCallback, useState } from 'react';
+import {
+  Button,
+  css,
+  Icon,
+  Label,
+  Link,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Radio,
+  RadioGroup,
+  spacing,
+} from '@mongodb-js/compass-components';
+import {
+  closeExportModal,
+  selectCurrentModel,
+  getCurrentDiagramFromState,
+} from '../store/diagram';
+import { connect } from 'react-redux';
+import type { DataModelingState } from '../store/reducer';
+import type { StaticModel } from '../services/data-model-storage';
+import { exportToJson } from '../services/export-diagram';
+
+const nbsp = '\u00a0';
+
+const modelBodyStyles = css({
+  paddingTop: spacing[600],
+});
+
+const exportFormatContainerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[300],
+});
+
+const radioGroupStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[300],
+});
+
+const footerStyles = css({
+  display: 'flex',
+  gap: spacing[200],
+});
+
+type ExportDiagramModalProps = {
+  isModalOpen: boolean;
+  diagramLabel: string;
+  model: StaticModel | null;
+  onCloseClick: () => void;
+};
+
+const ExportDiagramModal = ({
+  isModalOpen,
+  diagramLabel,
+  model,
+  onCloseClick,
+}: ExportDiagramModalProps) => {
+  const [exportFormat, setExportFormat] = useState<'json' | null>(null);
+
+  const onExport = useCallback(() => {
+    if (!exportFormat || !model) {
+      return;
+    }
+    exportToJson(diagramLabel, model);
+    onCloseClick();
+  }, [exportFormat, onCloseClick, model, diagramLabel]);
+
+  return (
+    <Modal
+      open={isModalOpen}
+      setOpen={onCloseClick}
+      data-testid="export-diagram-modal"
+    >
+      <ModalHeader
+        title="Export data model"
+        subtitle={
+          <div>
+            Export the data modal to JSON format.
+            {nbsp}
+            <Link
+              href="https://www.mongodb.com/docs/manual/data-modeling//"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              Learn more
+            </Link>
+          </div>
+        }
+      />
+      <ModalBody className={modelBodyStyles}>
+        <div className={exportFormatContainerStyles}>
+          <Label htmlFor="">Select file format:</Label>
+          <RadioGroup
+            className={radioGroupStyles}
+            value={exportFormat}
+            onChange={(e) => setExportFormat(e.target.value as 'json')}
+          >
+            <Radio value="json" aria-label="JSON">
+              <Icon glyph="CurlyBraces" />
+              {nbsp}
+              JSON
+            </Radio>
```

Comment:
This doesn't match the designs where the icon is outside of the radio item, not sure if that's a deliberate choice not to follow the designs here. Also the alignments of items inside the label is slightly off, but catches the eye, in the designs everything is middle aligned, not baseline

Reply:
I missed it as I was reusing some of the code from poc. I aligned it with the design in [6c41226](https://github.com/mongodb-js/compass/pull/7046/commits/6c41226177f9b389e61b89ab451690c58647f8a3)

Reply:
This is how it looks now

<img width="230" alt="image" src="https://github.com/user-attachments/assets/5841801b-6790-4815-b58d-35fcfd5f040f" />